NYPLAudiobookToolkit/Network/AudiobookNetworkService.swift

```diff
@@ -79,6 +80,18 @@ public final class DefaultAudiobookNetworkService: AudiobookNetworkService {
         return taskCompletedPercentage / Float(self.spine.count)
     }
   
+    /// Since download progress would be 0 if the download has not begun,
+    /// we use this computed property to check if the files have been downloaded or not.
+    public var downloadCompleted: Bool {
+        var completed = true
+        self.spine.forEach { spineElement in
+            if !spineElement.downloadTask.downloadCompleted {
+                completed = false
```

Comment:
minor, but efficiency wise once you set the completed flag to `false`, we can break and return right away. However I don't recall if you can do that with a `forEach`, you might need a C-style `for` or `while` loop.

This doesn't change the worst case complexity, but it does change the best/avg case.

Reply:
Fixed, I was getting build error when I `return` within `forEach`, didn't know it's not allowed in `forEach`